compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1593,7 +1593,8 @@ impl<'tcx> Place<'tcx> {
     }
 
     /// If MirPhase >= Derefered and if projection contains Deref,
-    /// It's guaranteed to be in the first place
+    /// It's guaranteed to be in the first place, this functions checks
+    /// both of these conditions and must only be called while MirPhase >= Derefered
```

Comment:
Both of which conditions? And it only does that check in debug-mode, so really this is *not* checked most of the time! The caller can *not* rely on this check.

Reply:
Also there is no phase called "Derefered"? At least look at https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/syntax/enum.MirPhase.html I cannot find one.

Reply:
My bad, it should been MirPass not MirPhase, https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_mir_transform/lib.rs.html#454-463

Reply:
Passes get reordered though. MirPhase is the right concept to refer to here. MirPhase make certain guarantees. For example, the `PostCleanup` one makes the guarantee we care about here. So that's what it should reference.